mmf/models/transformers/backends/huggingface.py

```diff
@@ -0,0 +1,218 @@
+                )
+                # Add random normal noise
+                self.token_type_embeddings.weight.data[idx] += torch.normal(
+                    self.model_config.token_noise_mean,
+                    self.model_config.token_noise_std,
+                    size=self.token_type_embeddings.weight.data[idx].size(),
+                )
+
+    def forward(
+        self,
+        input_ids: Dict[str, Tensor],
+        position_ids: Dict[str, Tensor],
+        segment_ids: Dict[str, Tensor],
+    ) -> Tensor:
+        list_embeddings = []
+        for idx, (token_emb, pos_emb, layer_norm, dropout) in enumerate(
+            zip(
+                self.token_embeddings,
+                self.pos_embeddings,
+                self.layer_norms,
+                self.dropouts,
+            )
+        ):
+            modality_name = self.modality_keys[idx]
+            total_embedding = token_emb(input_ids[modality_name])
+            if modality_name in position_ids:
+                total_embedding += pos_emb(position_ids[modality_name])
+
+            if modality_name in segment_ids:
+                total_embedding += self.token_type_embeddings(
+                    segment_ids[modality_name]
+                )
+
+            list_embeddings.append(dropout(layer_norm(total_embedding)))
+
+        return torch.cat(list_embeddings, dim=1)
+
+
+class HuggingfaceBackend(BaseTransformerBackend):
+    """Transformer backend wih Huggingface transformer models
+    """
+
+    def __init__(self, config: BaseTransformerConfigType, *args, **kwargs):
+        super().__init__(config)
+
+        # Replace transformer layers with scriptable JIT layers
+        replace_with_jit()
+
+    def build_transformer_config(self):
+        self.transformer_config = AutoConfig.from_pretrained(
+            self.config.transformer_base, **OmegaConf.to_container(self.config)
+        )
+
+    def build_transformer_base(self):
+        self.transformer = AutoModel.from_pretrained(
+            self.config.transformer_base, config=self.transformer_config
+        )
+
+    def build_embeddings(self):
+        self.embeddings = HFEmbeddings(
+            self.config, self.transformer_config, self.transformer
+        )
+
+    def get_config(self):
+        return self.transformer_config
+
+    def _transformer_encode(self, embedding, attention_mask):
+        if torch.jit.is_scripting():
+            return self.transformer.encoder(embedding, attention_mask)
+        else:
+            return self.transformer.encoder(
+                embedding, attention_mask, [None] * len(self.transformer.encoder.layer)
+            )
+
+    def forward(
+        self,
+        tokens_ids: Dict[str, Tensor],
+        position_ids: Dict[str, Tensor],
+        segment_ids: Dict[str, Tensor],
+        masks: List[Tensor],
+    ) -> Tuple[Tensor, List[Tensor]]:
+        # Transformer Input Embeddings
+        embedding_output = self.embeddings(
```

Comment:
This should be moved under `generate_embeddings` function where each backend do more stuff on their embeddings output (transpose etc.) if needed.